File: 9. Palindrome Number.py
def isPalindrome(x):
    if x < 0 or x % 10 == 0:
        return False
    res = 0
    while x > res:
        res = res * 10 + x % 10
        x = x // 10
    return ((res == x) or (res//10 == x))


if __name__ == '__main__':
    print('{} is {}'.format(13579, isPalindrome(13579)))
    print('{} is {}'.format(12321, isPalindrome(12321)))
    print('{} is {}'.format(128123321821, isPalindrome(128123321821)))

File: cppversion/9. Palindrome Number.cpp
#include <iostream>

bool isPalindrome(int x);

int main()
{
    // int a, b, c;
    std::cout << "a=12345 is " << isPalindrome(12345);
    std::cout << "\na=12321 is " << isPalindrome(12321);
    std::cout << "\na=123321 is " << isPalindrome(123321);
    std::cin.get();
}

bool isPalindrome(int x)
{
    if (x < 0 || x % 10 == 0)
        return false;
    int res = 0;
    while (x > res)
    {
        res = res * 10 + x % 10;
        x = x / 10;
    }
    return ((x == res) || (x == res / 10));
}